src/assets/keycloak/keycloak.md

```diff
@@ -0,0 +1,68 @@
+---
+title: keycloak
+authorName: Amrita Shrestha
+authorAvatar: https://avatars.githubusercontent.com/u/54478846?s=100&v=4
+authorLink: https://github.com/amrita-shrestha
+createdAt: March 26, 2024
+tags: Keycloak
+banner: https://blog.jankaritech.com/src/assets/keycloak/images/keycloak.png
+---
+
+## Keycloak
+Keycloak is an [open-source](https://github.com/keycloak/keycloak) project for Red Hat SSO. It provides `Identity and Access Management` solution designed to secure application services.
+Additionally, it enables users to authenticate through various identity providers and use fine-grained permissions for regulating access to Software as a Service (SaaS) applications.
+Keycloak alternatives are Auth0, Zluri, Microsoft Azure Active Directory, Okta, WSO2 Identity Server.
+
+Keycloak facilitates the creation of a user database with customizable roles and groups, offering functionalities such as user management, registration, and password policy enforcement.
+This makes it a comprehensive Identity and Access Management (IAM) solution for contemporary applications. Keycloak's API enables the integration of all these features into your application
+without the need for additional coding.
+
+Despite its extensive features, Keycloak does have certain limitations. One of these is the need for a more varied implementation approach.
+Additionally, the following are some of the drawbacks associated with Keycloak:
+1. `Complex Server Deployment:` The manual process involved in deploying Keycloak on a server can be complicated, potentially affecting overall productivity.
+2. `Lack of Comprehensive Support Documentation:` Despite its robust functionalities, Keycloak could benefit from more efficient and extensive support documentation. This would make it easier for users to find suitable solutions to their problems.
+
+## Keycloak Features
+![Keycloak features](/src/assets/keycloak/images/keycloak-features.png)
+
+1. `Identity Broker:`
+   Identity Broker acts as a middleman, linking various service providers with different identity providers. It establishes trust with external identity providers to utilize their identities for accessing internal services offered by service providers.
+   Imagine textile company,which has two keycloak setups: one for customer and another for internal employees. If you want your employees to access external services without creating new accounts on customer keycloak, you can use the customer Keycloak
+   as the mediator and the customer Keycloak as the source in this scenario.
+
+2. `User federation:`
```

Comment:
```suggestion
2. `User Federation:`
```
There are two ways for spelling titles in English: either all words are capitalized or only the first word (unless there are names). Try to be consistent in which style you apply throughout your blog. You can also ask the English teacher for recommendations on this.